RMQClient/RMQConnection.m

```diff
@@ -321,17 +321,33 @@ - (void)start {
 }
 
 - (void)close {
-    for (RMQOperation operation in self.closeOperations) {
+    [self givingCallingCloseWhileDisconnectedErrorIfNeeded];
+    for (RMQOperation operation in self.safeCloseOperations) {
```

Comment:
@tonyli508 should we enqueue any operations if we previously invoked an error callback on the delegate here? Other clients usually throw exceptions but this is not how errors are communicated in this case. WDYT?

Reply:
I guess this depends on business decision as well. If you say user need to call close after connection made, then encounter error should just do nothing. If close has higher priority then should just close. In my opinion close has higher priority.

Reply:
That's a good point. In other words we should try to close the connection because that's the user's intent. I agree.